src/messages.cs

```diff
@@ -0,0 +1,261 @@
+// Authors:
+//	Vincent Dondain <vincent@xamarin.com>
+//
+// Copyright 2016 Xamarin Inc. All rights reserved.
+//
+
+using System;
+using XamCore.CoreFoundation;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+#if !MONOMAC
+namespace XamCore.Messages {
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Native]
+	public enum MSMessagesAppPresentationStyle : nuint
+	{
+		Compact,
+		Expanded
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Native]
+	public enum MSStickerSize : nint
+	{
+		Small,
+		Regular,
+		Large
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[BaseType (typeof(UIViewController))]
+	interface MSMessagesAppViewController
+	{
+		[NullAllowed, Export ("activeConversation", ArgumentSemantic.Strong)]
+		MSConversation ActiveConversation { get; }
+
+		[Export ("presentationStyle", ArgumentSemantic.Assign)]
+		MSMessagesAppPresentationStyle PresentationStyle { get; }
+
+		[Export ("requestPresentationStyle:")]
+		void RequestPresentationStyle (MSMessagesAppPresentationStyle presentationStyle);
+
+		[Export ("dismiss")]
+		void Dismiss ();
+
+		[Export ("willBecomeActiveWithConversation:")]
+		void WillBecomeActive (MSConversation conversation);
+
+		[Export ("didBecomeActiveWithConversation:")]
+		void DidBecomeActive (MSConversation conversation);
+
+		[Export ("willResignActiveWithConversation:")]
+		void WillResignActive (MSConversation conversation);
+
+		[Export ("didResignActiveWithConversation:")]
+		void DidResignActive (MSConversation conversation);
+
+		[Export ("willSelectMessage:conversation:")]
+		void WillSelectMessage (MSMessage message, MSConversation conversation);
+
+		[Export ("didSelectMessage:conversation:")]
+		void DidSelectMessage (MSMessage message, MSConversation conversation);
+
+		[Export ("didReceiveMessage:conversation:")]
+		void DidReceiveMessage (MSMessage message, MSConversation conversation);
+
+		[Export ("didStartSendingMessage:conversation:")]
+		void DidStartSendingMessage (MSMessage message, MSConversation conversation);
+
+		[Export ("didCancelSendingMessage:conversation:")]
+		void DidCancelSendingMessage (MSMessage message, MSConversation conversation);
+
+		[Export ("willTransitionToPresentationStyle:")]
+		void WillTransition (MSMessagesAppPresentationStyle presentationStyle);
+
+		[Export ("didTransitionToPresentationStyle:")]
+		void DidTransition (MSMessagesAppPresentationStyle presentationStyle);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[BaseType (typeof(NSObject))]
+	interface MSConversation
+	{
+		[Export ("localParticipantIdentifier")]
+		NSUuid LocalParticipantIdentifier { get; }
+
+		[Export ("remoteParticipantIdentifiers")]
+		NSUuid[] RemoteParticipantIdentifiers { get; }
+
+		[NullAllowed, Export ("selectedMessage")]
+		MSMessage SelectedMessage { get; }
+
+		[Export ("insertMessage:localizedChangeDescription:completionHandler:")]
+		void InsertMessage (MSMessage message, [NullAllowed] string changeDescription, [NullAllowed] Action<NSError> completionHandler);
```

Comment:
The completion handler (here and the next API) makes it looks like it's missing an `[Async]` ?